starter-core/src/test/groovy/io/micronaut/starter/template/TomlTemplateSpec.groovy

```diff
@@ -0,0 +1,30 @@
+package io.micronaut.starter.template
+
+import io.micronaut.starter.feature.config.Configuration
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+
+class TomlTemplateSpec extends Specification {
+    private static String toml(Configuration configuration) {
+        def template = new TomlTemplate(null, configuration)
+        def bos = new ByteArrayOutputStream()
+        template.write(bos)
+        return new String(bos.toByteArray(), StandardCharsets.UTF_8)
+    }
+
+    def 'simple'() {
+        given:
+        def config = new Configuration('', '', '')
+        config.put('foo.bar', 4)
+        config.put('foo.baz', ['foo': 123, 'bar': [1, 2, ['name': 'fox']]])
+
+        expect:
+        toml(config) == '''foo.bar = 4
```

Comment:
I would recommend to replace:

```groovy
        toml(config) == '''foo.bar = 4
```
with: 

```
        toml(config) == '''\
foo.bar = 4
```

see: https://groovy-lang.org/syntax.html#_triple_single_quoted_string